openforcefield/topology/molecule.py

```diff
@@ -4592,70 +4593,95 @@ def add_conformer(self, coordinates):
+            - rdkit
+            - openeye
+            - nglview (conformers needed)
+        
+        Returns
+        -------
+        object
+            Depending on the backend chosen:
+            - rdkit, openeye -> IPython.display.Image
+            - nglview -> nglview.NGLWidget
+        """
         from openforcefield.utils.toolkits import OPENEYE_AVAILABLE, RDKIT_AVAILABLE
 
         backend = backend.lower()
 
         if backend == 'nglview':
-            import nglview as nv
             if self.conformers:
-                traj = _OFFTrajectoryNGLView(self)
-                widget = nv.NGLWidget()
-                widget.add_trajectory(traj)
+                import nglview as nv
+                trajectory_like = _OFFTrajectoryNGLView(self)
+                widget = nv.NGLWidget(trajectory_like)
                 return widget
             else:
                 raise ValueError('Visualizing with NGLview requires a molecule has conformers.')
-        elif backend == 'rdkit':
-            if RDKIT_AVAILABLE:
-                from rdkit.Chem.Draw import IPythonConsole
-                return self.to_rdkit()
-        elif backend == 'openeye':
-            if OPENEYE_AVAILABLE:
-                from openeye import oedepict
-                from IPython.display import Image
-
-                oemol = self.to_openeye()
-
-                width, height = 500, 300
-                opts = oedepict.OE2DMolDisplayOptions(width, height, oedepict.OEScale_AutoScale)
-
-                oedepict.OEPrepareDepiction(oemol)
-                img = oedepict.OEImage(width, height)
-                display = oedepict.OE2DMolDisplay(oemol, opts)
-                oedepict.OERenderMolecule(img, display)
-                png = oedepict.OEWriteImageToString("png", img)
-                return Image(png)
+        elif backend == 'rdkit' and RDKIT_AVAILABLE:
+            from rdkit.Chem.Draw import IPythonConsole
+            return self.to_rdkit()
+        elif backend == 'openeye' and OPENEYE_AVAILABLE:
+            from openeye import oedepict
+            from IPython.display import Image
+
+            oemol = self.to_openeye()
+
+            width, height = 500, 300
+            opts = oedepict.OE2DMolDisplayOptions(width, height, oedepict.OEScale_AutoScale)
+
+            oedepict.OEPrepareDepiction(oemol)
+            img = oedepict.OEImage(width, height)
+            display = oedepict.OE2DMolDisplay(oemol, opts)
+            oedepict.OERenderMolecule(img, display)
+            png = oedepict.OEWriteImageToString("png", img)
+            return Image(png)
         else:
             raise ValueError('Could not find an appropriate backend')
 
-# TODO: Gracefull skip this class if nglview is not installed
-import nglview as nv
-import uuid
-class _OFFTrajectoryNGLView(nv.Trajectory):
+
+try:
+    from nglview import Trajectory as _NGLViewTrajectory
+except ImportError:
+    _NGLViewTrajectory = object
```

Comment:
We still want the full module to be importable even when `nglview` is not there. `_OFFTrajectoryNGLView` will only be called when we try to visualize something with nglview, so maybe we can skip the alternative definition... I just don't feel comfortable with defining classes only when certain conditions are met, so I'd rather have the empty shell than nothing at all. But this is only a personal opinion, so please let's discuss this with the rest of the team! I don't have any strong opinions!

Reply:
That's a good point, I didn't even think of the fact that this would mean the module is not importable without `nglview` installed. I changed the exception to be raised only when the class would be instantiated, not when it's defined, so that should only be hit if a user is trying to use it.